src/sentry/integrations/tasks/sync_status_inbound.py

```diff
@@ -0,0 +1,261 @@
+                            )
+                            # Find the next release after the current_release_version
+                            # i.e. the release that resolves the issue
+                            resolved_in_release = (
+                                Release.objects.filter(
+                                    date_order_q,
+                                    projects=group.project,
+                                    organization_id=organization_id,
+                                )
+                                .extra(select={"sort": "COALESCE(date_released, date_added)"})
+                                .order_by("sort", "id")[:1]
+                                .get()
+                            )
+                            resolution_params.update({"release": resolved_in_release})
+                            activity_data.update({"version": resolved_in_release.version})
+                            local_logging_params.update(
+                                {"resolved_in_release": resolved_in_release.version}
+                            )
+                            logger.info(
+                                "get_resolutions_and_activity_data_for_groups.resolved_in_release",
+                                extra=local_logging_params,
+                            )
+                        except Release.DoesNotExist:
+                            # If it gets here, it means we don't know the upcoming
+                            # release yet because it does not exist, and so we should
+                            # fall back to our current model
+                            logger.info(
+                                "get_resolutions_and_activity_data_for_groups.next_release_not_found",
+                                extra=local_logging_params,
+                            )
+
+                resolutions_by_group_id[group.id] = resolution_params
+
+    return resolutions_by_group_id, activity_type, activity_data
+
+
+@instrumented_task(
+    name="sentry.integrations.tasks.sync_status_inbound",
+    queue="integrations",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+    silo_mode=SiloMode.REGION,
+)
+@retry(exclude=(Integration.DoesNotExist,))
+@track_group_async_operation
+def sync_status_inbound(
+    integration_id: int, organization_id: int, issue_key: str, data: Mapping[str, Any]
+) -> None:
+    from sentry.integrations.mixins import ResolveSyncAction
+
+    integration = integration_service.get_integration(integration_id=integration_id)
+    if integration is None:
+        raise Integration.DoesNotExist
+
+    organizations = Organization.objects.filter(id=organization_id)
+    affected_groups = Group.objects.get_groups_by_external_issue(
+        integration, organizations, issue_key
+    )
+    if not affected_groups:
+        return
+
+    installation = integration.get_installation(organization_id=organization_id)
```

Comment:
Would it be bad, to leave off the type hint for the return type of `get_installation()` ? Trying to import `IssueSyncMixin` in the `get_installation()` file (to typehint, IntegrationInstallation | IssueSyncMixin) causes an explosion of circular import errors. Currently alot of the mypy errors are solved by hasattr() which, doesn't seem as clean as saying,  installation is X | Y type.